### PR TITLE
ddl: mitigate the issue that multiple owner might exist during force to be owner

### LINKED ISSUE #56924: upgrade from 7.5.3 to 8.4.0 failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run sysbench read write
2、upgrade from 7.5.3 to 8.4.0
[tidb-0.log](https://github.com/user-attachments/files/17551671/tidb-0.log)
[tidb-1.log](https://github.com/user-attachments/files/17551676/tidb-1.log)

### 2. What did you expect to see? (Required)
upgrade succeed

### 3. What did you see instead (Required)
upgrade blocked due to tidb can not start

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v7.5.3
Edition: Community
Git Commit Hash: 70bfd90035cc81e80d78b4f1f76ca4baef8a5756
Git Branch: HEAD
UTC Build Time: 2024-07-31 13:11:26
GoVersion: go1.21.10
Race Enabled: false
Check Table Before Drop: false
Store: unistore

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #56924, #56713

Problem Summary:

### What changed and how does it work?
older versions which don't have the fix of https://github.com/pingcap/tidb/issues/54689, when upgrade and forcing newer version node to be owner, multiple owner might exist at the same time, this pr tries to mitigate the issue, see code comments for more detail

also mitigate https://github.com/pingcap/tidb/issues/57003 by sleep after force to be owner

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - start a v7.5.0 cluster with 3 TiDBs, rolling upgrade to this pr, success
  - start a v8.1.1 cluster with 3 TiDBs, rolling upgrade to this pr, success
  - start a v7.5.0 cluster with 3 TiDBs, stop all TiDBs, then start them with this pr, success
    - **Note: after this PR, we will aquire distributed lock before start domain, so in this case, the 2 non-owner node will not involve in schema version sync for upgrade DDL**
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/session/tidb.go [hunk 1]
```diff
@@ -125,6 +125,8 @@ var (
 		domains: map[string]*domain.Domain{},
 	}
 	// store.UUID()-> IfBootstrapped
+	// TODO this memory flag is meaningless, a store is always bootstrapped once,
+	// we can always get the version from the store, remove it later.
```
Comment (yudongusa) [inline pkg/session/tidb.go:129]: How can we ensure this "TODO" can be tracked and done "later"?

File: pkg/ddl/ddl.go [hunk 2]
```diff
@@ -93,6 +93,20 @@ var (
 	detectJobVerInterval = 10 * time.Second
 )
 
+// StartMode is an enum type for the start mode of the DDL.
+type StartMode string
+
+const (
+	// Normal mode, cluster is in normal state.
+	Normal StartMode = "normal"
```
Comment (Benjamin2037) [inline pkg/ddl/ddl.go:101]: What is difference between normal mode with bootstrap mode? I mean whatever update or normal situation, we both need bootstrap.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]

**Pull request merged**